server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -223,24 +223,26 @@ public void unmarkSegmentToDrop(DataSegment dataSegment)
 
   private void processSegmentChangeRequest()
   {
-    if (currentlyProcessing != null) {
-      log.debug(
-          "Server[%s] skipping processSegmentChangeRequest because something is currently loading[%s].",
-          basePath,
-          currentlyProcessing.getSegmentIdentifier()
-      );
+    synchronized (lock) {
```

Comment:
I think it won't work, as all the threads will be checking `currentlyProcessing` until the first thread will reset it (set it to `null`), which will be after the historical node will take and confirm the segment.

Reply:
Got it, initially, I thought you'd like to parallelize processing within a single queue peon also.